### PR TITLE
Create s3/session test

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR implements internal s3/session.go test.
It includes the refactoring on FOP base on the coding guideline

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by hlts2 — DISMISSED** [review]
LGTM

File: internal/db/storage/blob/s3/session/session_test.go [hunk 1]
```diff
@@ -47,36 +100,57 @@ func TestNew(t *testing.T) {
...
+					enableParamValidation:      true,
+					enable100Continue:          true,
+					enableContentMD5Validation: true,
+					enableEndpointDiscovery:    false,
+					enableEndpointHostPrefix:   true,
+				},
+			},
+		},
+		{
+			name: "return and log session when set option error",
```
Comment (vankichi) [inline internal/db/storage/blob/s3/session/session_test.go:125]:
```suggestion
			name: "return default option and logging session error when set option error",
```
How about this?

Comment (vankichi) [reply internal/db/storage/blob/s3/session/session_test.go:125]: I think if you can it'd be better adding a test of set option with no error.

**Review by vankichi — COMMENTED** [review]
@kevindiu leave comments

File: internal/db/storage/blob/s3/session/session_test.go [hunk 2]
```diff
@@ -47,36 +100,80 @@ func TestNew(t *testing.T) {
...
+					enableParamValidation:      true,
+					enable100Continue:          true,
+					enableContentMD5Validation: true,
+					enableEndpointDiscovery:    false,
+					enableEndpointHostPrefix:   true,
+				},
+			},
+		},
+		{
+			name: "return default option and logging session error when set option error",
```
Comment (vankichi) [inline internal/db/storage/blob/s3/session/session_test.go:148]:
```suggestion
			name: "return session with default option and logging session error when set option error",
```

**Review by vankichi — APPROVED** [review]
LGTM

**Review by hlts2 — DISMISSED** [review]
LGTM

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by vankichi — DISMISSED** [review]

**Review by hlts2 — APPROVED** [review]

**Review by vankichi — APPROVED** [review]

**Pull request merged**